### ISSUE TITLE
Diagram preview

### ISSUE THREAD

**Issue by ctkqiang** [gabeidx/zed-plantuml#3]
Hey @gabeidx , 

 Currently, the installed PlantUML package in Zed editor does not support live visualization or rendering of PlantUML diagrams. This lack of visual feedback makes it difficult to validate and review diagrams directly within the editor.

**Expected Behavior**
- A live visualization of the PlantUML diagram should appear in a split view or dedicated panel within Zed.
- The diagram should update dynamically as the .puml file is edited.

**Actual Behavior**

- No live visualization or rendering is available.
- No option to preview the PlantUML diagram directly in Zed.

**Comment by Vaneeza-7** [comment]
Yeah. This is much needed. Would make things a lot easier.

**Comment by gabeins** [comment]
Hello there! 

I absolutely want to have diagram preview working via the extension. 

I asked around in Zed's Discord to see if there's a built-in way to do it, but alas, there isn't one currently. The closest feature–that opens a panel with something rendered other than source code–is the "View release notes" and that uses their internal Markdown renderer.

What is needed is to figure out a way to wrap PlantUML into an LSP and use that in the extension to trigger things from Zed events. Only then a workaround would be viable, for example, saving the output to a temp folder and asking Zed to open it. I don't _love_ this idea, but it's possible.

I'm open to different approaches.

**Comment by gabeins** [comment]
I'm having difficulty finding time to work on this at the moment.

@ctkqiang @Vaneeza-7 since you have expressed interest in the feature, would any of you like to contribute?

**Comment by ctkqiang** [comment]
> I'm having difficulty finding time to work on this at the moment.
> 
> [@ctkqiang](https://github.com/ctkqiang) [@Vaneeza-7](https://github.com/Vaneeza-7) since you have expressed interest in the feature, would any of you like to contribute?

Of course, I’d love to help out! But first, I’ll need to learn how to get started with this for Zed. If you can point me to any resources, or docs, or give me a quick idea of what’s needed, I’ll dive right in!

**Comment by ctkqiang** [comment]
> Update 👍🏼  : PR [https://github.com/gabeidx/zed-plantuml/pull/4](https://github.com/gabeidx/zed-plantuml/pull/4)

**Comment by stalexsm** [comment]
https://crates.io/crates/plantuml-server-client-rs

**Comment by morgenthum** [comment]
@gabeidx can we merge this? I would **love** to use this feature.

**Comment by gabeins** [comment]
> [@gabeidx](https://github.com/gabeidx) can we merge this? I would **love** to use this feature.

As it stands, the PR can not be merged. All review comments remain unaddressed.

**Comment by gabeins** [comment]
Today [Zed Preview 0.194.0](https://zed.dev/releases/preview/0.194.0) was released with a feature that may make this feature easier to implement: https://github.com/zed-industries/zed/pull/32694

**Comment by gavin-hu** [comment]
Diagram preview +1
